### PR TITLE
mono_native_tls_get_value that does not change LastError.

### PR DESCRIPTION
**Pull request by jaykrell**
Should fix https://github.com/mono/mono/issues/15541
without the overhead of https://github.com/mono/mono/pull/15465.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/utils/mono-tls.h [hunk 1]
```diff
@@ -52,7 +52,32 @@ g_static_assert (TLS_KEY_DOMAIN == 0);
...
+//
+#include <winternl.h>
+
+static inline
+void*
+mono_native_tls_get_value (unsigned index)
+{
+	PTEB const teb = NtCurrentTeb ();
+
+	g_assert (index < TLS_MINIMUM_AVAILABLE + 1024);
```
Comment (filipnavara) [inline mono/utils/mono-tls.h]: I see you changed this... I assume `TLS_EXPANSION_SLOTS` was undefined so you replaced it with 1024. Perhaps `8 * sizeof(NtCurrentTeb()->Peb->TlsExpansionBitmapBits)` or something similar is an option?

Comment (jaykrell) [reply mono/utils/mono-tls.h]: It is defined but only in obscure headers. It wasn't compiling.
winternal.h does not reveal this either.

Comment (jaykrell) [reply mono/utils/mono-tls.h]: i.e. Peb->Bitmap.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
Let's remove the assert about 1024 actually, in case the limit is ever raised.

File: mono/utils/mono-tls.h (hunk 1 above)
Comment (jaykrell) [reply mono/utils/mono-tls.h]: Let's remove it anyway, in case limit is raised.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**